common/meta/api/src/schema_api_test_suite.rs

```diff
@@ -424,6 +425,140 @@ impl SchemaApiTestSuite {
         Ok(())
     }
 
+    pub async fn database_rename<MT: SchemaApi>(self, mt: &MT) -> anyhow::Result<()> {
+        let tenant = "tenant1";
+        let db_name = "db1";
+        let db2_name = "db2";
+        let new_db_name = "db3";
+
+        tracing::info!("--- rename not exists db1 to not exists db2");
+        {
+            let req = RenameDatabaseReq {
+                if_exists: false,
+                name_ident: DatabaseNameIdent {
+                    tenant: tenant.to_string(),
+                    db_name: db_name.to_string(),
+                },
+                new_db_name: new_db_name.to_string(),
+            };
+
+            let res = mt.rename_database(req).await;
+            tracing::info!("rename database res: {:?}", res);
+            assert!(res.is_err());
+            assert_eq!(
+                ErrorCode::UnknownDatabase("").code(),
+                ErrorCode::from(res.unwrap_err()).code()
+            );
+        }
+
+        tracing::info!("--- prepare db1 and db2");
+        {
+            // prepare db2
+            let res = self.create_database(mt, tenant, "db1", "eng1").await?;
+            assert_eq!(1, res.db_id);
+
+            tracing::info!("--- rename not exists db4 to exists db1");
+            {
+                let req = RenameDatabaseReq {
+                    if_exists: false,
+                    name_ident: DatabaseNameIdent {
+                        tenant: tenant.to_string(),
+                        db_name: "db4".to_string(),
+                    },
+                    new_db_name: db_name.to_string(),
+                };
+
+                let res = mt.rename_database(req).await;
+                tracing::info!("rename database res: {:?}", res);
+                assert!(res.is_err());
+                assert_eq!(
+                    ErrorCode::UnknownDatabase("").code(),
+                    ErrorCode::from(res.unwrap_err()).code()
+                );
+            }
+
+            // prepare db2
+            let res = self.create_database(mt, tenant, "db2", "eng1").await?;
+            assert!(res.db_id > 1);
+        }
+
+        tracing::info!("--- rename exists db db1 to exists db db2");
+        {
+            let req = RenameDatabaseReq {
+                if_exists: false,
+                name_ident: DatabaseNameIdent {
+                    tenant: tenant.to_string(),
+                    db_name: db_name.to_string(),
+                },
+                new_db_name: db2_name.to_string(),
+            };
+
+            let res = mt.rename_database(req).await;
+            tracing::info!("rename database res: {:?}", res);
+            assert!(res.is_err());
+            assert_eq!(
+                ErrorCode::DatabaseAlreadyExists("").code(),
+                ErrorCode::from(res.unwrap_err()).code()
+            );
+        }
+
+        tracing::info!("--- rename exists db db1 to not exists mutable db");
```

Comment:
Because we has some immutable database like system and default. So I think other database that can be created or dropped is mutable database.

Like immutable catalog and mutable catalog.